File: C_C++/셀프넘버.cpp
#include <iostream>
#define limit 10001

using namespace std;

int main() {
	bool arr[limit] = {false};
	for (int i = 1; i < limit; i++) {
		int result=i;
		for (int j=i; j > 0; j /= 10) {
		
			result += (j % 10);
		}
		if (result <limit) arr[result] = true;
	}
	for (int i = 0; i < limit; i++) {
		if(arr[i]==false)	cout << i<<"\n";
	}
	return 0;
}

File: python/난이도 쉬움/셀프넘버.py


if __name__=="main":
    limit =10001

    arr=[False for i in range(limit)]

    for number in range(1,limit):
        result=number
        while number>0:
            result+=number%10
            number=number//10
        if result<limit:  arr[result] =True
    for i in range(1,limit):
        if arr[i]==False: print(i)
    